From bc760f26f85204b377701537f74078717b318a9b Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Mon, 14 Nov 2022 13:39:39 -0600
Subject: [PATCH 0755/1793] windowscodecs: HACK: Avoid using
 __builtin_frame_address() for setjmp().

This is a workaround to compiler bug (observed with Mingw / Gcc 10.x).

CW-Bug-Id: #21563
---
 dlls/windowscodecs/wincodecs_private.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlls/windowscodecs/wincodecs_private.h b/dlls/windowscodecs/wincodecs_private.h
index 9da717fc345..76db7439db1 100644
--- a/dlls/windowscodecs/wincodecs_private.h
+++ b/dlls/windowscodecs/wincodecs_private.h
@@ -398,4 +398,9 @@ extern HRESULT CommonDecoder_CreateInstance(struct decoder *decoder,
 extern HRESULT CommonEncoder_CreateInstance(struct encoder *encoder,
     const struct encoder_info *encoder_info, REFIID iid, void** ppv);
 
+#ifdef _WIN64
+#undef setjmp
+#define setjmp(buf) _setjmpex(buf, NULL)
+#endif
+
 #endif /* WINCODECS_PRIVATE_H */
-- 
2.46.0

